infrastructure/lib/constructs/database.ts

```diff
@@ -82,13 +88,13 @@ export class Database extends cdk.Construct {
   // Using a ConcreteDependable means that we can change the underlying definition of what it
   // means for this to be "ready" without having to update the API. We just need to add/change
   // resources in the ConcreteDependable.
```

Comment:
I suggest updating the name to avoid any confusion. Looking at the cdk docs for `DependencyGroup` it does not appear to have much of a difference (other than accepting an array of dependencies) and it appears your comment still holds true with the name change.
```suggestion
  // Using a DependencyGroup means that we can change the underlying definition of what it
  // means for this to be "ready" without having to update the API. We just need to add/change
  // resources in the DependencyGroup.
```